pkgs/applications/science/machine-learning/vowpal-wabbit/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, stdenv, fetchgit, cmake, boost169, zlib }:
+
+let
+  version = "8.9.2";
+in
+stdenv.mkDerivation {
+  pname = "vowpal-wabbit";
+  version = "${version}";
+  src =
+    fetchgit {
+      url = "https://github.com/VowpalWabbit/vowpal_wabbit.git";
+      rev = "${version}";
+      sha256 = "0np11nvxvbxgfdlm4q07lw7zni6mm4l93bf7pj0rif7fz15k8f3l";
+      fetchSubmodules = true;
+    };
+  buildInputs = [
+    cmake
```

Comment:
`cmake` should be in `nativeBuildInputs`, this will automatically do the stuff in your `configure`, `build` and `install` phases.

To add cmake flags, please use the `cmakeFlags` array.